api/src/main/java/org/openmrs/BaseOpenmrsObject.java

```diff
@@ -79,6 +82,11 @@ public boolean equals(Object obj) {
 		if (!(obj instanceof BaseOpenmrsObject)) {
 			return false;
 		}
+		//In case of hibernate proxy objects we need to get real classes
+		Class<?> thisClass = Hibernate.getClass(this);

```

Comment:
It seems fine and should work :) however i am not not sure :) Please Correct if i am wrong